test/deploy/rackhd_microservice_install.py

```diff
@@ -0,0 +1,185 @@
+    - loads prerequisite packages docker
+    - loads prerequisite packages docker-compose
+    - downloads RackHD dell microservices packages
+    - installs using docker
+    - load configuration files
+    - startup and verify operations
+
+usage:
+    python run_tests.py -stack <stack ID> -test deploy/rackhd_microservice_install.py
+'''
+
+import fit_path  # NOQA: unused import
+import fit_common
+import flogging
+import os
+import socket
+import time
+import unittest
+
+logs = flogging.get_loggers()
+
+# sets up proxy if required
+PROXYVARS = ''
+if fit_common.fitproxy()['host'] != '':
+    # note that both proxy server settings below are set to the same http: URL
+    PROXYVARS = "export http_proxy=http://" + fit_common.fitproxy()['host'] + ":" + fit_common.fitproxy()['port'] + ";" + \
+                "export https_proxy=http://" + fit_common.fitproxy()['host'] + ":" + fit_common.fitproxy()['port'] + ";"
+
+
+class rackhd_micro_install(unittest.TestCase):
+    def test01_update_sudoers_info(self):
+        # update sudoers to preserve proxy environment
+        logs.info(" ***** Update sudoers proxy env ****")
+        sudoersproxy = open("sudoersproxy", 'w')
+        sudoersproxy.write('Defaults env_keep="HOME http_proxy https_proxy ftp_proxy"\n')
+        sudoersproxy.close()
+        fit_common.remote_shell('pwd')
+        fit_common.scp_file_to_ora("sudoersproxy")
+        self.assertEqual(fit_common.remote_shell('cp sudoersproxy /etc/sudoers.d/'
+                                                 )['exitcode'], 0, "sudoersproxy config failure.")
+        os.remove('sudoersproxy')
+
+    def test02_add_nfs_mount(self):
+        logs.info(" ***** Add NFS mount point ****")
+        self.assertEqual(fit_common.remote_shell(PROXYVARS + "apt-get -y install nfs-kernel-server")['exitcode'],
+                         0, "nfs-kernel-server install failure.")
+        nfsexports = open("nfsexports", 'w')
+        nfsexports.write('/nfs               *(rw,sync,no_subtree_check,no_root_squash,no_all_squash)\n')
+        nfsexports.close()
+        fit_common.scp_file_to_ora("nfsexports")
+
+        # Making a backup copy of /etc/exports file
+        rc = fit_common.remote_shell('test -e /etc/microservices_exports.bak')
+
+        if rc['exitcode'] == 0:
+            fit_common.remote_shell('cp /etc/microservices_exports.bak /etc/exports')
+        else:
+            fit_common.remote_shell('cp /etc/exports /etc/microservices_exports.bak')
+
+        self.assertEqual(fit_common.remote_shell('cat nfsexports >> /etc/exports'
+                                                 )['exitcode'], 0, "nfsexports config failure.")
+
+        # Make an nfs mount point
+        self.assertEqual(fit_common.remote_shell('sudo mkdir -p /nfs'
+                                                 )['exitcode'], 0, "Failed to mkdir /nfs")
+        # Ubuntu 14.04 nfs services
+        self.assertEqual(fit_common.remote_shell('sudo service nfs-kernel-server start'
```

Comment:
Parts of this script are specific to Ubuntu 14.04 commands, should we name the script specific to that or add comments in the script header?
Or, you can determine the ubuntu version running via cat /etc/lsb-release and issue appropriate format for version 14 vs 16.